src/pineko/check.py

```diff
@@ -65,3 +66,32 @@ def check_grid_and_eko_compatible(pineappl_grid, operators, xif):
     # x-grid
     if not np.all(in1d(np.unique(operators["targetgrid"]), np.array(x_grid))):
         raise ValueError("x grid in pineappl grid and eko operator are NOT compatible!")
+
+def check_grid_contains_fact_sv(grid_path):
+    """Checks whether factorization scale-variations are available in the pineappl grid.
+    Parameters
+    ----------
+        grid_path : pathlib.Path
+            path to grid
+    """
+    pineappl_grid = pineappl.grid.Grid.read(grid_path)
```

Comment:
please change the argument to be the object (instead of the path)

Reply:
Yes, better to operate on objects, and load the grid only ones.
Otherwise you slow down everything, and it would make much harder to use for other functions (what should I do if I have an object? dump it and pass a path? much better the opposite: if it accepts object, and you have a path, just load it).